public/app.js

```diff
@@ -21,69 +29,101 @@ let createSelect = () => {
   })
 }
 createSelect()
+// .2 Remove selection
+
+// .3 Display selected description
+
+//Reset the game
+const resetGame = () =>{
+    window.location.reload();
+    return false
+}
+// Reset highscore
+const resetHighScore = () =>{
+    highScoreStorage.setItem('highScore', 0);
+    highScoreDisplay.innerHTML = highScoreStorage.getItem('highScore')
 
+}
+
+//Generate computer choice
 const generateComputerChoice = () => {
     computer.choice = choices[Math.floor(Math.random() * choices.length)]
     return computer.choice
 }
 const getPlayerChoice = () => {
     let selectedValue = document.querySelector(".selected")
+    if (player.choice == null) {
+        null
+    }else{
+
     player.choice = selectedValue.children[0].getAttribute('alt');
+    }
+    
     return player.choice
 }
-// const selectWinner = () =>{
-
-//     generateComputerChoice();
-//     getPlayerChoice();
 
-//     console.log(player.choice+computer.choice)
-// }
+//Compare choices/selectwinner
 const selectWinner = () =>{
     let result = null;
     generateComputerChoice();
     getPlayerChoice();
 
-    switch(player.choice+computer.choice){
-        //Win choice
-        case "firewood":
-        case "firemetal":
-        case "earthfire":
-        case "earthwater":
-        case "woodearth":
-        case "woodwater":
-        case "metalwood":
-        case "metalearth":
-        case "waterfire":
-        case "watermetal":
-            result = "won";
-            break;
-        //Lose choice
-        case "firewater":
-        case "fireearth":
-        case "earthwood":
-        case "earthmetal":
-        case "woodfire":
-        case "woodmetal":
-        case "metalfire":
-        case "metalwater":
-        case "waterwood":
-        case "waterearth":
-            result = "losed";
-            break;
-        //Draw choice
-        case "firefire":
-        case "earthearth":
-        case "woodwood":
-        case "metalmetal":
-        case "waterwater":
-            result = "draw";
-            break;
+    if (player.choice !== null) {
+        switch(player.choice+computer.choice){
```

Comment:
Ten switch koniecznie musi zostać zmieniony. Opieranie się na połączonych stringach nie wygląda jak dobry pomysł. Metoda może być bardzo ciężka w utrzymaniu w przyszłym developmencie. Generalnie korzystanie ze switch/case w js jest już trochę traktowane jako antypattern, są lepsze metody na osiągnięcie tego samego wyniku.

Moja propozycja realizacji tablicy wyników: utworzenie pliku `pairs.json`, w którym będziesz trzymał obiekt, zawierający wszystkie wybory użytkownika jako klucze i obiekt z wyborami komputera jako wartości.

```json
{
   "fire": {
      "wood": true,
      "water": false
   }
}
```

i stworzyć metodę do określania wyników, która weźmie ten obiekt i zwróci wynik w formie boola:
```ts
import pairs from 'pairs.json';

const getResult = (userChoice: string; computerChoice: string): boolean => pairs[userChoice][computerChoice];
```